lib/active_record/connection_adapters/odbc_adapter.rb

```diff
@@ -773,36 +774,34 @@ def add_limit_offset!(sql, options)
           if offset = options[:offset] then sql << " OFFSET #{offset}" end
         end
 
+        def generate_result_set(query_result=[])
+          query_result ||= []
+          fields = query_result[:column_descriptors].map{|x| activeRecIdentCase(x.name)}
+
+          values = []
+          query_result[:rows].each do |row|
+            values << row.map { |val| convertOdbcValToGenericVal(val) }
+          end
```

Comment:
What you have is already a big improvement over what this was refactored from. Much easier to understand.